eng/common/TestResources/Get-TestSecrets.ps1

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env pwsh
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+#Requires -Modules @{ModuleName='Az.Accounts'; ModuleVersion='1.6.4'}
+#Requires -Modules @{ModuleName='Az.KeyVault'; ModuleVersion='4.6.0'}
+
+[CmdletBinding()]
+[OutputType([pscustomobject])]
+param (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string] $VaultName,
+
+    [Parameter(Mandatory=$true, Position=1)]
+    [string] $Name
```

Comment:
I think as a follow-on we can add multi vault and multi secret handling in order to support other yaml/test configs. I have some helpers for subscription config object merging that we can re-use for this: https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/SubConfig-Helpers.ps1#L93

Reply:
I can do that now, but give me a bit. I was doing this on the side and need to focus on a few things. Should have an update by tomorrow.

Reply:
Another thing I realized, which I'm happy to do as another follow-on: To support places where people need values from a cloud config that also contains subscrption and service principal values, we could support a switch like `-UseServicePrincipal`,  defaulting to false, which would strip out the `*ApplicationId` and `*ApplicationSecret` etc. values, so that we can still default to using the user's credentials when running locally.

Reply:
If you wanted to contribute that to this PR as well as what we talked about above, that might be good. I've been busy readying some packages for next week so this has been lower pri.

Reply:
@heaths I have some time today. Want me to take on multi vault/secret support as well or do you have changes pending for that?

Reply:
Feel free. I don't have anything pending right now. If you don't have time, I will have a little next week.

Reply:
@benbp did we still want to update this? I forgot all about it and, even after reading the comments above, aren't quite paging in the context.